app/models/collection_draft.rb

```diff
@@ -30,24 +30,25 @@ def get_next_form(name, direction)
     end
 
     def create_from_collection(collection, user, native_id)
+      # TODO: try to refactor this method for all drafts
       new_entry_title = (collection['EntryTitle'].blank?) ? nil : collection['EntryTitle']
 
       if native_id
         # Edited record
-        draft = CollectionDraft.find_or_create_by(native_id: native_id)
+        draft = self.find_or_create_by(native_id: native_id)
```

Comment:
so just using "self" worked well, no need to subclass or anything these, like the solution.